docs/FirestoreSPM.md

```diff
@@ -0,0 +1,152 @@
+`Package.swift` manifest. The setup is designed to support two different build
+options for Firestore: from source or from a pre-compiled binary. This choice is
+controlled by the `FIREBASE_SOURCE_FIRESTORE` environment variable.
+
+## Main Product
+
+The main entry point for integrating Firestore via Swift Package Manager is the
+`FirebaseFirestore` library product.
+
+```swift
+.library(
+  name: "FirebaseFirestore",
+  targets: ["FirebaseFirestoreTarget"]
+)
+```
+
+This product points to a wrapper target, `FirebaseFirestoreTarget`, which then
+depends on the appropriate Firestore targets based on the chosen build option.
+
+---
+
+## Wrapper Target
+
+The `FirebaseFirestoreTarget` is a thin wrapper that exists to work around a
+limitation in Swift Package Manager where a single target cannot conditionally
+depend on different sets of targets (source vs. binary).
+
+By having clients depend on the wrapper, the `Package.swift` can internally
+manage the complexity of switching between source and binary builds based on the
+`FIREBASE_SOURCE_FIRESTORE` environment variable. This provides a stable entry
+point for all clients and avoids pushing conditional logic into their own
+package manifests.
+
+---
+
+## 1. Binary-Based Build
+
+When the `FIREBASE_SOURCE_FIRESTORE` environment variable is **not** set (which is
+the default), Swift Package Manager will use pre-compiled binaries for Firestore
+and its heavy dependencies.
+
+### Dependency Hierarchy
+
+The dependency tree for a binary-based build is as follows:
+
+```
+FirebaseFirestore (Library Product)
+└── FirebaseFirestoreTarget (Wrapper Target)
+    └── FirebaseFirestore (Swift Target)
+        ├── FirebaseAppCheckInterop
+        ├── FirebaseCore
+        ├── FirebaseCoreExtension
+        ├── FirebaseSharedSwift
+        ├── leveldb
+        ├── nanopb
+        ├── abseil (binary) (from https://github.com/google/abseil-cpp-binary.git)
+        ├── gRPC-C++ (binary) (from https://github.com/google/grpc-binary.git)
+        └── FirebaseFirestoreInternalWrapper (Wrapper Target)
+            └── FirebaseFirestoreInternal (Binary Target)
+```
+
+### Target Breakdown
+
+*   **`FirebaseFirestoreTarget`**: A wrapper target, same as in the source-based
+    build.
+*   **`FirebaseFirestore`**: The Swift target containing the public API. In this
+    configuration, it depends on the binary versions of abseil and gRPC, as
+    well as the `FirebaseFirestoreInternalWrapper`.
+*   **`FirebaseFirestoreInternalWrapper`**: A thin wrapper target that exists to
+    expose the headers from the underlying binary target.
+*   **`FirebaseFirestoreInternal`**: This is a `binaryTarget` that downloads and
+    links the pre-compiled `FirebaseFirestoreInternal.xcframework`. This
+    framework contains the compiled C++ core of Firestore.
+
+---
+
+## 2. Source-Based Build
+
+When the `FIREBASE_SOURCE_FIRESTORE` environment variable is set, Firestore and
```

Comment:
Should we document how to set `FIREBASE_SOURCE_FIRESTORE` since, correct me if I'm wrong, this can only be set before launching Xcode. I haven't tried setting `FIREBASE_SOURCE_FIRESTORE` in the Environment Variables for the scheme and then running "Reset Package Caches" but I'm fairly sure that wouldn't work.

Reply:
That's a great suggestion! Will do.